XRTK.SDK/Packages/com.xrtk.sdk/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -1,22 +1,137 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.Rendering;
+using UnityEngine.Serialization;
+using XRTK.Attributes;
 using XRTK.Definitions.Utilities;
 using XRTK.EventDatum.Input;
+using XRTK.Extensions;
 using XRTK.Interfaces.InputSystem;
 using XRTK.Interfaces.InputSystem.Handlers;
 using XRTK.SDK.Input.Handlers;
 using XRTK.Services;
 
 namespace XRTK.SDK.UX
 {
-    public class BoundingBox : BaseFocusHandler,
+    public class BoundingBox : MonoBehaviour
+    {
+        private class BoundingBoxRig : BaseInputHandler,
             IMixedRealitySourceStateHandler,
-            IMixedRealityInputHandler,
+            IMixedRealityPointerHandler,
             IMixedRealityInputHandler<MixedRealityPose>
-    {
+        {
+            public override bool IsFocusRequired => false;
+
+            public BoundingBox BoundingBoxParent
+            {
+                get
+                {
+                    if (boundingBoxParent == null)
+                    {
+                        Destroy(gameObject);
+                    }
+
+                    return boundingBoxParent;
+                }
+                set => boundingBoxParent = value;
+            }
+
+            private BoundingBox boundingBoxParent;
+
+            /// <inheritdoc />
+            void IMixedRealityPointerHandler.OnPointerDown(MixedRealityPointerEventData eventData)
```

Comment:
Public or Private? Please state

Reply:
It's explicit. Interface implementations are always public.

Reply:
Except this is Private :S 
```
void IMixedRealityPointerHandler
```